tools/distrib/check_copyright.py

```diff
@@ -105,10 +131,23 @@
 
 RE_YEAR = r'Copyright (?P<first_year>[0-9]+\-)?(?P<last_year>[0-9]+) ([Tt]he )?gRPC [Aa]uthors(\.|)'
 RE_LICENSE = dict(
-    (k, r'\n'.join(LICENSE_PREFIX[k] +
+    (k, r'\n'.join(LICENSE_PREFIX_RE[k] +
                    (RE_YEAR if re.search(RE_YEAR, line) else re.escape(line))
                    for line in LICENSE_NOTICE))
-    for k, v in LICENSE_PREFIX.items())
+    for k, v in LICENSE_PREFIX_RE.items())
+
+import datetime
+
+YEAR = datetime.datetime.now().year
+
+LICENSE_YEAR = f'Copyright {YEAR} gRPC authors.'
+LICENSE_TEXT = dict((k, '\n'.join(
+    ([LICENSE_PREFIX_TEXT[k][0]] if LICENSE_PREFIX_TEXT[k][0] else []) + [
+        LICENSE_PREFIX_TEXT[k][1] + ' ' +
+        (LICENSE_YEAR if re.search(RE_YEAR, line) else line)
+        for line in LICENSE_NOTICE
+    ] + ([LICENSE_PREFIX_TEXT[k][2]] if LICENSE_PREFIX_TEXT[k][2] else [])))
+                    for k, v in LICENSE_PREFIX_TEXT.items())
```

Comment:
I known this is concatenating the license string. If we use intermediate variables, it would help other people to understand.

Reply:
I've extracted the logic to a separate function. Hope this makes it a bit more clear.